src/client.rs

```diff
@@ -108,12 +110,64 @@ impl Client {
             let token = self.auth_token.clone().unwrap();
             headers.insert(
                 reqwest::header::AUTHORIZATION,
-                format!("OAuth {}", token).parse().unwrap(),
+                format!("OAuth {}", token).parse()?,
             );
         }
 
         let response = self.http_client.get(url).headers(headers).send().await?;
-        response.error_for_status()
+        response.error_for_status().map_err(Error::from)
+    }
+
+    pub fn get_stream<T>(&self, path: &str, num_pages: u64) -> BoxStream<Result<T>>
+    where
+        T: DeserializeOwned + 'static + Send,
+    {
+        unfold(self.clone(), self.get_pages(&path), num_pages)
+    }
+
+    fn get_pages<T>(&self, path: &str) -> BoxFuture<Result<Page<T>>>
+    where
+        T: DeserializeOwned + 'static + Send,
+    {
+        self.get_page(&(self.host.clone() + path))
+    }
+
+    fn get_pages_url<T>(&self, url: &str) -> BoxFuture<Result<Page<T>>>
+    where
+        T: DeserializeOwned + 'static + Send,
+    {
+        self.get_page(url)
+    }
+
+    fn get_page<T>(&self, path: &str) -> BoxFuture<Result<Page<T>>>
+    where
+        T: DeserializeOwned + 'static + Send,
+    {
+        let mut url = Url::parse(path).unwrap();
+
+        if !url.query_pairs().any(|(q, _)| q == "client_id") {
+            url.query_pairs_mut()
+                .append_pair("client_id", &self.client_id);
+        }
+
+        let mut headers = reqwest::header::HeaderMap::new();
+
+        if self.auth_token.is_some() {
+            let token = self.auth_token.clone().unwrap();
+            headers.insert(
+                reqwest::header::AUTHORIZATION,
+                format!("OAuth {}", token).parse().unwrap(),
+            );
+        }
```

Comment:
You don't need to clone the token here. The following should work, if not you can try `self.auth_token.as_ref()` instead of `Some(ref token)`.

```suggestion
        if let Some(ref token) = self.auth_token {
            headers.insert(
                reqwest::header::AUTHORIZATION,
                format!("OAuth {}", token).parse().unwrap(),
            );
        }
```